plugins/dev-create/templates/block/template/test/block_test.mocha.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {assert} from 'chai';
+import {sinon} from 'sinon';
+const {BlockTemplate} = require('../dist/index');
+
+suite('BlockTemplate', function() {
+  setup(function() {
+    this.workspace = new Blockly.Workspace();
+    this.block = this.workspace.newBlock('block_template');
+  });
+
+  teardown(function() {
+    this.workspace.dispose();
+  });
+
+  test('Structure', function() {
+    // TODO
+    // Example:
+    // assert.exists(this.block.inputList, 'Has inputList');
+    // assert.equal(this.block.inputList.length, 3);
+    // assert.equal(this.block.getInput('IF').connection.check_.length, 1);
+    // assert.equal(this.block.getInput('IF').connection.check_[0], 'Boolean');
+    // assert.exists(this.block.onchangeWrapper_, 'Has onchange handler');
+  });
+
+  suite('blockToCode', function() {
+    // TODO comment out supported generators
+    const supportedGenerators = [
+      // [Blockly.Dart, 'Dart'],
+      [Blockly.JavaScript, 'JavaScript'],
+      // [Blockly.Lua, 'Lua'],
+      // [Blockly.PHP, 'PHP'],
+      // [Blockly.Python, 'Python'],
+    ];
+
+    const assertGeneratedCodeEqual = (generator, block, expectedCode) => {
+      const code = generator.blockToCode(block);
+      assert.equal(code, expectedCode);
+    };
+
+    suite('Trivial', function() {
+      supportedGenerators.forEach(function(generatorInfo) {
+        const generator = generatorInfo[0];
+        const name = generatorInfo[1];
+        test(name, function() {
+          const expectedCode = ''; // TODO update with expected code
+          assertGeneratedCodeEqual(generator, this.block, expectedCode);
+        });
+      });
+    });
+
+    // TODO add other relevant tests, such as with connected inputs
+  });
+
+  suite('Serialization', function() {
+    test('blockToXml', function() {
+      const expectedXml = ''; // TODO
+      const xml = Blockly.Xml.domToText(Blockly.Xml.blockToDom(this.block));
+      assert.equal(xml, expectedXml);
+    });
+
+    test('xmlToBlock', function() {
+      const block = Blockly.Xml.domToBlock(Blockly.Xml.textToDom(
+          '<block type="block_template"/>'
+      ), this.workspace);
+      // TODO
+    });
+
+    // TODO see block_test
```

Comment:
What is this referring to?

Reply:
This is more a note for myself block_test in core. I plan on removing it.